src/compact_strings.rs

```diff
@@ -489,17 +489,18 @@ impl Index<usize> for CompactStrings {
 /// assert_eq!(iter.next(), Some("Three"));
 /// assert_eq!(iter.next(), None);
 /// ```
-pub struct Iter<'a> {
-    inner: &'a CompactStrings,
-    iter: core::slice::Iter<'a, Metadata>,
-}
+pub struct Iter<'a>(crate::compact_bytestrings::Iter<'a>);
 
 impl<'a> Iter<'a> {
-    #[inline]
     pub fn new(inner: &'a CompactStrings) -> Self {
-        Self {
-            inner,
-            iter: inner.0.meta.iter(),
+        Self(inner.0.iter())
+    }
+
+    fn from_utf8_maybe_checked(bytes: &[u8]) -> Option<&str> {
```

Comment:
Since this is an internal method, I think it would make sense to just call it `from_utf8` as there should be no invariants to uphold for the end user.